### PR TITLE
Add and constrain derivative functions in state machines

### PR DESCRIPTION
**Pull request by andreas-junghanns**

### CODE REVIEW

**Comment by andreas-junghanns** [comment]
@KarlWernersson : added as requested.

@MBlesken & @IZacharias : I have removed the use of fmi3GetAdjointDerivative and fmi3GetDirectionalDerivative from Initialization Mode for SE - is that correct? (How would one compute any derivative in SE?)

**Review by KarlWernersson — APPROVED** [review]
Looks good. Thanks for fixing it =)

**Comment by MBlesken** [comment]
> @MBlesken & @IZacharias : I have removed the use of fmi3GetAdjointDerivative and fmi3GetDirectionalDerivative from Initialization Mode for SE - is that correct? (How would one compute any derivative in SE?)

@andreas-junghanns @IZacharias I'm not sure if this should be forbidden. On the other hand I don't see why one would use it in SE. Maybe this is enough reason to forbid it...?

File: docs/2_3_common_states.adoc [hunk 1]
```diff
@@ -390,9 +390,11 @@ This function should always be called after calling function <<fmi3UpdateDiscret
 
 Function <<fmi3GetDirectionalDerivative>>::
 See <<fmi3GetDirectionalDerivative>>.
+This call is not allowed in Scheduled Execution.
```
Comment (MBlesken) [inline docs/2_3_common_states.adoc:393]: @andreas-junghanns @IZacharias I'm not sure if this should be forbidden. On the other hand I don't see why one would use it in SE. Maybe this is enough reason to forbid it...?

File: docs/2_3_common_states.adoc [hunk 2]
```diff
@@ -390,9 +390,11 @@ This function should always be called after calling function <<fmi3UpdateDiscret
...
 
 Function <<fmi3GetAdjointDerivative>>::
 See <<fmi3GetAdjointDerivative>>.
+This call is not allowed in Scheduled Execution.
```
Comment (MBlesken) [inline docs/2_3_common_states.adoc:397]: ... same here

**Pull request merged**